### PR TITLE
[MOBILE-1739] Update Android flutter plugin to new api version

### PR DESCRIPTION
**Pull request by Apekka**
### What do these changes do?
These changes upgrade the Flutter Android plugin to new API version

### Why are these changes necessary?
To solve an issue reported by a customer
https://github.com/urbanairship/airship-flutter/issues/56

### How did you verify these changes?
By running the app, and testing the basic functionalities of Airship.

### CODE REVIEW

File: android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt [hunk 1]
```diff
@@ -110,7 +116,12 @@ class FlutterInboxMessageView(private var context: Context, channel: MethodChann
     }
 }
 
-class AirshipPlugin : MethodCallHandler {
+class AirshipPlugin : MethodCallHandler, FlutterPlugin, ActivityAware, ServiceAware {
```
Comment (rlepinski) [inline android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt:119]: do we need to implement ActivityAware and ServiceAware interfaces?

Comment (Apekka) [reply android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt:119]: In the Flutter docs, it says :
```
(Optional) If your plugin needs an Activity reference, also implement the ActivityAware interface.

(Optional) If your plugin is expected to be held in a background Service at any point in time, implement the ServiceAware interface.
```

And if I'm not wrong, I guess the Plugin interacts with the Activity, and is expected to be held in a background Service.
However, looks like everything is working fine even when I remove ActivityAware and ServiceAware. So I guess I will remove them

File: android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt [hunk 2]
```diff
@@ -110,18 +112,32 @@ class FlutterInboxMessageView(private var context: Context, channel: MethodChann
...
 
+    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel = MethodChannel(binding.flutterEngine.dartExecutor, "com.airship.flutter/airship")
+        binding.platformViewRegistry.registerViewFactory("com.airship.flutter/InboxMessageView", InboxMessageViewFactory(binding))
+        channel.setMethodCallHandler(this)
+        context = binding.applicationContext
+    }
+
+    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
+        channel.setMethodCallHandler(null)
```
Comment (marc-scig) [inline android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt:138]: should we also release the other references here (channel, context)?

Comment (Apekka) [reply android/src/main/kotlin/com/airship/flutter/AirshipPlugin.kt:138]: Yeah I released the channel here, but I don't think that releasing the context is needed.
However, I may be wrong

**Pull request merged**